backbone.js

```diff
@@ -1140,6 +1140,7 @@
         removed.push(model);
         this._removeReference(model, options);
       }
+      if (models.length > 0 && !options.silent) delete options.index;
```

Comment:
> Why the options.silent check? If it's just to check whether add will fire, shouldn't we always remove it?

If options.silent == true, then options.index was never set. The check saves the expensive delete operation.

> Also, you don't have to delete this prop (delete is slow). Just set it to undefined.

Since you are passing options to an event callback, you don't know how the callback will iterate over the options object.  If you set it to `void 0` and the event callback does a `for(key in options)`, then `options.index` will still show up, `options.hasOwnProperty('index') == true`.  

The event callbacks could then get any of the following cases:
1. `options.index = value`
2. `options.index = undefined`
3. `options` does not own property `index`

I'm not dogmatic about it, I'll make the change if you are ok sacrificing consistency on the last two cases for performance. I think you can make a reasonable argument that the callbacks should be able to handle all 3 cases.  Then it would make sense to remove the options.silent check.